mitodl/odl-video-service
# Enabled multiple edX endpoints for posting HLS videos

#### What are the relevant tickets?
Closes #841 

#### What's this PR do?
- Allows collections to be configured such that uploaded video files can be sent to multiple edX instances after they're HLS-encoded
- If the correct settings exist, creates an app default edX instance to which all HLS-encoded videos will be posted (this mirrors existing behavior)

#### How should this be manually tested?
Before running the app, make sure you have `EDX_BASE_URL` and `EDX_ACCESS_TOKEN` settings filled out. An access token can be created in your edX instance in Django admin: http://edx.odl.local:18000/admin/oauth2/accesstoken/

- After migrations run, make sure you have an `EdxEndpoint` record that matches your settings, and that `is_global_default` is set to True
- Upload a video to any collection. Confirm that after HLS-encoding a request was made to post the video to that default edX instance
- Add a new `EdxEndpoint`. It's OK to just use dummy values. Add that new `EdxEndpoint` to some `Collection` in Django admin
- Upload another video. Confirm that after HLS-encoding a request was made to post the video to both the default edX endpoint _and_ the endpoint that you just added.
- Also try out the `add_hls_video_to_edx` to make sure it still works as expected.

#### Any background context you want to provide?
This PR does not include a UI to add/edit these endpoints for a collection in the front-end. This decision was made because (a) we haven't thought at all about what the design should be for this, (b) it's a pretty sizable chunk of work to add this UI (there isn't even a select component in this codebase yet), and (c) adding/editing edX endpoints for collections is not likely to occur very frequently, and should be manageable via Django admin for at least a short time.

#### Screenshots (if appropriate)
![ss 2020-03-24 at 17 16 56 ](https://user-images.githubusercontent.com/14932219/77605813-3e03e000-6eec-11ea-88a5-135254d23e0d.png)
